src/main/kotlin/com/target/liteforjdbc/Db.kt

```diff
@@ -118,20 +118,34 @@ open class Db(
      * or rolls back if it throws an exception. This is required to perform any DB interactions that need transaction
      * support.
      */
-    fun <T> withTransaction(block: (Transaction) -> T): T {
+    fun <T> withTransaction(
+        isolationLevel: IsolationLevel,
+        block: (Transaction) -> T
+    ): T {
         val transaction = Transaction(connection = dataSource.connection)
-        transaction.use {
+        val currentIsolationLevel = transaction.connection.transactionIsolation
+        return transaction.use {
             try {
+                transaction.connection.transactionIsolation = isolationLevel.intCode
                 val result = block(transaction)
                 transaction.commit()
-                return result
+                result
             } catch (t: Throwable) {
                 transaction.rollback()
                 throw t
+            } finally {
+                transaction.connection.transactionIsolation = currentIsolationLevel
             }
         }
     }
 
+    /**
+     * Uses a com.target.liteforjdbc.Transaction, and commits it once to the block is executed successfully,
+     * or rolls back if it throws an exception. This is required to perform any DB interactions that need transaction
+     * support.
+     */
+    fun <T> withTransaction(block: (Transaction) -> T): T = withTransaction(IsolationLevel.TRANSACTION_READ_COMMITTED, block)
```

Comment:
I wonder if we should add the default isolation level to the configuration, and maybe let that be a default value of read committed. Then people could override the default value too. I'll defer to @chad-moller-target on this though.

Reply:
> I wonder if we should add the default isolation level to the configuration, and maybe let that be a default value of read committed. Then people could override the default value too. I'll defer to @chad-moller-target on this though.

If this is needed, should we do it in a separate PR, so that this one is small and atomic?

Reply:
> @AlexCue987 Because of the preferred parameter order putting the block on the end is still a breaking change. My gut instinct is to actually create this as a new overloaded endpoint. Then the existing endpoint can simply call withTransaction(IsolationLevel, block) with the default Isolation level.
> 
> Would this work?

Of course it will work. But I'm struggling to find a good name for the new function. Should it be withTransactionWithIsolation ? Can you suggest a better name?